src/sync.cpp

```diff
@@ -79,7 +79,14 @@ namespace librealsense
             LOG_WARNING(callbacks_inflight << " callbacks are still running on some other threads. Waiting until all callbacks return...");
         }
         // wait until user is done with all the stuff he chose to borrow
-        _callback_inflight.wait_until_empty();
+        try
+        {
+            _callback_inflight.wait_until_empty();
+        }
+        catch (...)
+        {
+            LOG_DEBUG("Error while waiting for an empty callback");
```

Comment:
Catch `const std::exception & e` and add `e.what()` to the log.
LOG_DEBUG("Error while waiting for user callback to finish: " + e.what());

Reply:
`+` sign did not pass compilation, I will use `<<`.